gdrapi.h

```diff
@@ -112,10 +117,10 @@ int gdr_map(gdr_t g, gdr_mh_t handle, void **va, size_t size);
 // First invoke gdr_unmap() then gdr_unpin_buffer().
 int gdr_unmap(gdr_t g, gdr_mh_t handle, void *va, size_t size);
 
-// gpubar_ptr is a user-space virtual address, i.e. one returned by gdr_map()
-int gdr_copy_to_bar(void  *gpubar_ptr, const void *cpumem_ptr, size_t size);
-int gdr_copy_from_bar(void *cpumem_ptr, const void *gpubar_ptr, size_t size);
-
+// map_d_ptr is the user-space virtual address belonging to a mapping of a device memory buffer,
+// i.e. one returned by gdr_map()
+int gdr_copy_to_mapping(gdr_mh_t handle, void *map_d_ptr, const void *h_ptr, size_t size);
```

Comment:
@pakmarkthub One thought I had is that we could in principle maintain some level of source code compatibility by maintaining the old APIs, i.e. `gdr_copy_to_bar()`, while introducing the new APIs, i.e. `gdr_copy_to_mapping`.
The old APIs would not leverage cached mappings, still that could be a good compromise.
Only when using the new APIs the user would experience the full performance improvement due to cached mappings on POWER9.